✨ Create new Rule | Do you use GitHub Discussions to suggest new Ideas? (#4817)
SSWConsulting/SSW.Rules.Content: rules/use-github-discussions/rule.md

---
type: rule
archivedreason: 

> title: Do you use Discussions to suggest new ideas?

Reviewer: For SEO
```suggestion
title: Do you use GitHub Discussions to suggest new ideas?
```

or 

```suggestion
title: GitHub - Do you use Discussions to suggest new ideas?
```

guid: 89c61225-1320-4a27-b9de-dafa54591a6f
uri: use-github-discussions

> created: 2022-01-25T09:31:0.0000000Z

Reviewer: 👀
```suggestion
created: 2023-01-25T09:31:0.0000000Z
```

related: []
redirects:
---

Discussions are a great place to chat with your team and brainstorm new ideas!

If you're thinking of making a Pull Request/Issue to suggest some changes to a document, code or something else. If you're not sure about your suggestions, you could leverage GitHub Discussions to see what people think!

Here's a [great video](https://www.youtube.com/watch?v=bErGYN3Ljz8) by the GitHub team that explains Discussions in full!

`youtube: https://www.youtube.com/embed/bErGYN3Ljz8`

**Video (40sec): What is GitHub Discussions?**

<!--endintro-->

When you're looking to make some changes, chat about the project or or otherwise engage with a project on GitHub. There are some simple steps you can take to choose the right place to engage. They are:

## Discussion to Issue

The default choice that one should make is to look for a discussion or create a discussion! If you're looking to create some new content for the repo, change some code or otherwise ask a question, discussions are a great place to start.

![Figure: SSW Rules | GitHub | Discussions](https://user-images.githubusercontent.com/66365977/214453285-b074f967-a637-4968-bd0d-ce79198f8bc3.png)

> From there you can add your comments and let others know about it by atting them (@BobNorthwind). When you're done discussing and you have something to action on the project, you can then easily convert the discussion into an issue!

Reviewer: ```suggestion
From there you can add your comments and let others know about it by @ mentioning them (E.g. @BobNorthwind). When you're done discussing and you have something to action on the project, you can then easily convert the discussion into an issue!
```

![Figure: SSW Rules | GitHub | Discussions - Convert a Discussion to an issue](https://user-images.githubusercontent.com/66365977/214461325-7bd47032-3b8f-4e1c-96f4-63b7ee02a64d.png)

> ![SSW Rules Content Discussion Converted To Issue](https://user-images.githubusercontent.com/66365977/214463615-2b27e427-93a9-4d49-ab2b-bdb06859c816.png)

Reviewer: ```suggestion
![Figure: SSW Rules Content Discussion Converted To Issue](https://user-images.githubusercontent.com/66365977/214463615-2b27e427-93a9-4d49-ab2b-bdb06859c816.png)
```

---
type: rule
archivedreason: 
title: Do you use Discussions to suggest new ideas?
guid: 89c61225-1320-4a27-b9de-dafa54591a6f
uri: use-github-discussions
created: 2022-01-25T09:31:0.0000000Z
authors:
- title: Sam Wagner
  url: https://ssw.com.au/people/sam-wagner
related: []
redirects:
---

Discussions are a great place to chat with your team and brainstorm new ideas!

If you're thinking of making a Pull Request/Issue to suggest some changes to a document, code or something else. If you're not sure about your suggestions, you could leverage GitHub Discussions to see what people think!

Here's a [great video](https://www.youtube.com/watch?v=bErGYN3Ljz8) by the GitHub team that explains Discussions in full!

`youtube: https://www.youtube.com/embed/bErGYN3Ljz8`

**Video (40sec): What is GitHub Discussions?**

<!--endintro-->

> When you're looking to make some changes, chat about the project or or otherwise engage with a project on GitHub. There are some simple steps you can take to choos the right place to engage. They are:

Author: ```suggestion
When you're looking to make some changes, chat about the project or or otherwise engage with a project on GitHub. There are some simple steps you can take to choose the right place to engage. They are:
```

## Discussion to Issue

The default choice that one should make is to look for a discussion or create a discussion! If you're looking to create some new content for the repo, change some code or otherwise ask a question, discussions are a great place to start.

> ![SSW Rules Content Discussions](https://user-images.githubusercontent.com/66365977/214453285-b074f967-a637-4968-bd0d-ce79198f8bc3.png)
> 
> **Figure: SSW Rules | GitHub | Discussions**

Author: ```suggestion
![Figure: SSW Rules | GitHub | Discussions](https://user-images.githubusercontent.com/66365977/214453285-b074f967-a637-4968-bd0d-ce79198f8bc3.png)
```

From there you can add your comments and let others know about it by atting them (@BobNorthwind). When you're done discussing and you have something to action on the project, you can then easily convert the discussion into an issue!

> ![SSW Rules Content Discussion convert to issue button](https://user-images.githubusercontent.com/66365977/214461325-7bd47032-3b8f-4e1c-96f4-63b7ee02a64d.png)
> 
> **Figure: SSW Rules | GitHub | Discussions - Convert a Discussion to an issue

Author: ```suggestion
![Figure: SSW Rules | GitHub | Discussions - Convert a Discussion to an issue](https://user-images.githubusercontent.com/66365977/214461325-7bd47032-3b8f-4e1c-96f4-63b7ee02a64d.png)
```

![SSW Rules Content Discussion Converted To Issue](https://user-images.githubusercontent.com/66365977/214463615-2b27e427-93a9-4d49-ab2b-bdb06859c816.png)

Now that it's an issue, you can of course associate Pull Requests with that issue and otherwise associate work with that rather than the whole project.

> Of these three stages: Discussion, Issue and Pull Request, you can always jump in at any point. For example, if all it is that you're doing is updating a spelling error. It might be worth just opening a Pull Request

Author: ```suggestion
Of these three stages: Discussion, Issue and Pull Request, you can always jump in at any point. For example, if all it is that you're doing is updating a spelling error, It might be worth just opening a Pull Request.
```
